wrapper/src/main/java/software/amazon/jdbc/plugin/failover2/FailoverConnectionPlugin.java

```diff
@@ -606,4 +621,70 @@ protected boolean canDirectExecute(final String methodName) {
         || methodName.equals(METHOD_IS_CLOSED)
         || methodName.equals(METHOD_ABORT));
   }
+
+  @Override
+  public Connection connect(
+      final String driverProtocol,
+      final HostSpec hostSpec,
+      final Properties props,
+      final boolean isInitialConnection,
+      final JdbcCallable<Connection, SQLException> connectFunc)
+      throws SQLException {
+    return connectInternal(driverProtocol, hostSpec, props, isInitialConnection, connectFunc, false);
+  }
+
+  private Connection connectInternal(String driverProtocol, HostSpec hostSpec, Properties props,
+      boolean isInitialConnection, JdbcCallable<Connection, SQLException> connectFunc, boolean isForceConnect)
+      throws SQLException {
+
+    if (props.containsKey(INTERNAL_CONNECT_PROPERTY_NAME)) {
```

Comment:
`forceConnect()` is for direct connection but it's not used in `failover2`. So `forceConnect()` makes not much sense to this case. Also, `failover2` uses regular `connect()` because `MonitoringHostListProvider` already verified topology so no need to force direct connect. And with a regular `connect()`, the call comes to `failover2` plugin again and our intention is to skip it.

Reply:
I see. Using the INTERNAL_CONNECT_PROPERTY_NAME property as a flag seems a bit hacky/strange to me but I can't think of an alternative at the moment.